app/models/couch.rb

```diff
@@ -16,6 +16,10 @@ def self.search(params)
       .distinct
   end
 
+  def available_nights
+    self.nights.map { |night| night.date if night.reservation_id.nil? }
```

Comment:
Well, this has to do with the javascript, I need the dates as an array of strings to feed into the jquery. This being used by datepicker to know which dates are available for selection.